npm/test/utils.ts

```diff
@@ -1,21 +1,34 @@
-import { KeycloakInstance } from "keycloak-js"
+import {KeycloakInstance, KeycloakTokenParsed} from "keycloak-js"
+import Mock = jest.Mock;
 
-export const mockKeycloakInstance: KeycloakInstance = {
-  init: jest.fn(),
-  login: jest.fn(),
-  logout: jest.fn(),
-  register: jest.fn(),
-  accountManagement: jest.fn(),
-  createLoginUrl: jest.fn(),
-  createLogoutUrl: jest.fn(),
-  createRegisterUrl: jest.fn(),
-  createAccountUrl: jest.fn(),
-  isTokenExpired: jest.fn(),
-  updateToken: jest.fn(),
-  clearToken: jest.fn(),
-  hasRealmRole: jest.fn(),
-  hasResourceRole: jest.fn(),
-  loadUserInfo: jest.fn(),
-  loadUserProfile: jest.fn(),
-  token: "fake-auth-token"
+export const createMockKeycloakInstance: (
+    updateToken?: Mock,
+    isTokenExpired?: Mock,
+    refreshTokenParsed?: KeycloakTokenParsed
+) => KeycloakInstance = (updateToken = jest.fn(),
+                         isTokenExpired = jest.fn(),
```

Comment:
All the places that use this parameter need to pass a function that returns a boolean. The type system doesn't help too much here because the type is just `Mock`.

Maybe the parameter could just be a simple boolean instead, and line 23 could be a function that returns the boolean value?

Reply:
Similarly for `refreshTokenParsed`. It looks like the main thing that needs to be passed as a parameter is the expiry time.

Reply:
Not sure changing the `refreshTokenParsed` simply to a date will work, as the test at line 230 requires no refresh token at all. So just passing a number won't work for that test.